lib/internal/event_target.js

```diff
@@ -228,33 +221,32 @@ class EventTarget {
     }
     type = String(type);
 
-    let root = this[kEvents].get(type);
+    const existingListener = this[kEvents].get(type);
 
-    if (root === undefined) {
-      root = { size: 1, next: undefined };
-      // This is the first handler in our linked list.
-      new Listener(root, listener, once, capture, passive);
-      this[kNewListener](root.size, type, listener, once, capture, passive);
-      this[kEvents].set(type, root);
+    if (existingListener === undefined) {
+      this[kNewListener](1, type, listener, once, capture, passive);
+      this[kEvents].set(type, new Listener(listener, once, capture, passive));
```

Comment:
might be over optimizing things, but could we skip the whole listener class? and use indexed items in the array? i.e.

```js
this[kEvents].set(type, [listener, once, capture, passive)])
```

To access the listener at location 2 you would do:

```js
const [listener, once, capture, passive] = this[kEvents].get(type)[index / 4]
```

Reply:
That would help a ton I think